python/revng/cli/_commands/idb_converter.py

```diff
@@ -0,0 +1,655 @@
+        return inf_structure.is_64bit()
+
+    def _collect_imports(self):
+        # NOTE: If @plt was used, the functions are in Functions() already, but the name starts
+        # with ".", and we do import types for them during parsing of regular/local functions.
+
+        # NOTE: We cannot use api.ida_nalt.get_import_module_qty() here, since it relies
+        # on `library names`, but IDA prints `.dynsym` as library for each library in case of ELF.
+
+        for mod_index in range(self.api.ida_nalt.get_import_module_qty() + 1):
+
+            def cb(function_addr, function_name, ignored_always_none):
+                assert ignored_always_none is None
+
+                function = idb.analysis.Function(self.idb, function_addr)
+                if not function:
+                    log_error(f"Unable to find function {function_name}.")
+                    return True
+
+                function_name = function.get_name()
+                function_type = None
+                try:
+                    function_type = function.get_signature()
+                except Exception as exception:
+                    log_error(f"Unable to parse function type for {function_name}.")
+                    log_error(str(exception))
+                    function_type = None
+                    return True
+
+                revng_function_type = None
+                if function_type is not None:
+                    qualified_revng_function_type = self._convert_idb_type_to_revng_type(
+                        function_type
+                    )
+                    revng_function_type = self.unwrap_qualified(qualified_revng_function_type)
+                else:
+                    abi = revng_arch_to_abiname[self.arch]
+
+                    qualified_return_type = self._get_primitive_type(m.PrimitiveTypeKind.Void, 0)
+                    revng_function_type = m.CABIFunctionType(
+                        OriginalName=function_name,
+                        ABI=abi,
+                        ReturnType=qualified_return_type,
+                        Arguments=[],
+                    )
+                    self.revng_types_by_id[revng_function_type.ID] = revng_function_type
+
+                dynamic_function = m.DynamicFunction(
+                    OriginalName=function_name,
+                    Prototype=m.Reference.create(m.Binary, revng_function_type),
+                )
+                self.dynamic_functions.append(dynamic_function)
+                return True
+
+            mod_name = self.api.ida_nalt.get_import_module_name(mod_index)
+            # NOTE: IDA 7 has a bug when reporting imported library names - prints `.dynsym` for
+            # each import.
+            if mod_name and mod_name != ".dynsym":
+                self.imported_libraries.append(mod_name)
+            self.api.ida_nalt.enum_import_names(mod_index, cb)
+
+    def _fixup_void_cabifunctiontype_arguments(self):
```

Comment:
Why? When does this happen?
Very mysterious.

Reply:
I thought that this is something that you discussed, but it makes no sense to me at all.

Reply:
I don't know, I suggest you put `assert False` above `argument.Type = self._get_primitive_type(m.PrimitiveTypeKind.Generic, 8)` and run the tests, see what triggers it and decide what to do.
If nothing triggers it, drop.

Reply:
I made a script that creates an idb file out of each ```.compiled``` test, and nothing triggers this, I am dropping it.